src/main/java/mat/client/shared/MeasureLibraryResultTable.java

```diff
@@ -107,6 +108,7 @@ public void run() {
                 } else if (obj.getClickCount() == 2  && obj.isFhirEditOrViewable()) {
                     singleClickTimer.cancel();
                     obj.setClickCount(0);
+                    MeasureComposerPresenter.getInstance().disableMeasurePackageTabForFhirMeasures(obj.getMeasureModel());
```

Comment:
It introduces a circular dependency... between the instances
can we try to utilize an event-based approach here?

You could try to subscribe MeasureComposerPresenter to MeasureSelectedEvent and MeasureEditEvent

Reply:
Implemented